### PR TITLE
[HW-2] Implement my own array methods

### PR DESCRIPTION
**Pull request by miseinen**

### CODE REVIEW

File: HW02/ekaterina_balabanovich/hw02.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+module Attempt
```
[the remark this answers is not in the document]
Comment (B0RlS) [reply HW02/ekaterina_balabanovich/hw02.rb:1]: Check out my advice in our telegram chat about commit names.

File: HW02/ekaterina_balabanovich/hw02.rb [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+  refine Array do
+    def my_map
+        mapped = []
```
Comment (B0RlS) [inline HW02/ekaterina_balabanovich/hw02.rb:4]:
```suggestion
        new_arr = []
```

File: HW02/ekaterina_balabanovich/hw02.rb [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+        for i in 0...self.length
+            mapped << yield(self[i]) if block_given?
```
Comment (B0RlS) [inline HW02/ekaterina_balabanovich/hw02.rb:6]: let's take this check out of the loop, there is no point in checking it at each step of the loop

**Comment by B0RlS** [comment]
Could you also change the file name please?
You should give the filename the same as the module name.

File: HW02/ekaterina_balabanovich/hw02.rb [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+        end
+        mapped
+    end
+
+    def my_select
+        selected = []
+        for i in 0...self.length
+            selected << self[i] if block_given? && yield(self[i])
```
Comment (B0RlS) [inline HW02/ekaterina_balabanovich/hw02.rb:14]: Maybe we should take this check out of the loop?

File: HW02/ekaterina_balabanovich/hw02.rb [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+        end
+        selected
+    end
+
+    def my_each
+        for i in 0...self.length
+            yield(self[i]) if block_given?
```
Comment (B0RlS) [inline HW02/ekaterina_balabanovich/hw02.rb:21]: same comment

File: .gitignore [hunk 6]
```diff
@@ -0,0 +1 @@
+HW02/ekaterina_balabanovich/.vscode/settings.json
```
Comment (vladhilko) [inline .gitignore:1]: Could you move this file inside your own folder `HW02/ekaterina_balabanovich` or just delete it?  You can also try to set your own global `.gitignore` https://sebastiandedeyne.com/setting-up-a-global-gitignore-file/

**Review by B0RlS — APPROVED** [review]
lgtm

**Pull request merged**